spark/src/main/scala/filodb.spark/FiloRelation.scala

```diff
@@ -65,21 +65,38 @@ object FiloRelation extends StrictLogging {
     }.toSeq
   }
 
-  // Single partition query?
-  def singlePartitionQuery(projection: RichProjection,
-                           filterStuff: Seq[(Int, KeyType, Seq[Filter])]): Option[Any] = {
+  // Partition Query?
+  def partitionQuery(config: Config,
+                     projection: RichProjection,
+                     filterStuff: Seq[(Int, KeyType, Seq[Filter])]): Seq[Any] = {
+    val inqueryPartitionsLimit = config.getInt("columnstore.inquery-partitions-limit")
     // Are all the partition keys given in filters?
-    // Are all the filters EqualTo?
-    val equalPredValues: Seq[Any] = filterStuff.collect {
-      case (pos, keyType, Seq(EqualTo(_, equalValue))) => KeyFilter.parseSingleValue(keyType)(equalValue)
+    // Are the filters EqualTo, In?
+    val predicateValues = filterStuff.collect {
+      case (pos, keyType, Seq(EqualTo(_, equalValue))) =>
+        Set(KeyFilter.parseSingleValue(keyType)(equalValue))
+      case (pos, keyType, Seq(In(_, inValues))) =>
+        (KeyFilter.parseValues(keyType)(inValues.toSet))
     }
-    if (equalPredValues.length == projection.partitionColumns.length) {
-      if (equalPredValues.length == 1) Some(equalPredValues.head) else Some(equalPredValues)
+    logger.info(s"Push down predicate values: ${predicateValues.toString()}")
+    if (predicateValues.length == projection.partitionColumns.length) {
+      if(predicateValues.length == 1) {
+        predicateValues.flatten
+      }
+      else {
+        val predList = combine(predicateValues)
+        if (predList.size <= inqueryPartitionsLimit) predList else Nil
+      }
     } else {
-      None
+      Nil
     }
   }
 
+  def combine[A](xs: Traversable[Traversable[A]]): Seq[Seq[A]] =
```

Comment:
I'm not convinced this does what you want.  

First of all, predicateValues might have the incorrect order if the order of the filters is not the same as the partition keys. For example, lets say your partition keys are countryCode, actorName.  If you do WHERE actorName = "blah" and countryCode IN (a, b), I think your filter will not work because the predicateValues would be Seq(Set("blah"), Set(a, b)), but you actually need it in countryCode and actorName order.   So you need to sort the predicateValues by position first.

Second, I now understand what you mean by generating combinations, but the code isn't too obvious.  Maybe add a comment to explain what is meant by combine?